cs/src/Management/TunnelManagementClient.cs

```diff
@@ -1337,6 +1352,15 @@ public void Dispose()
             return result;
         }
 
+        /// <summary>
+        /// Event fired when a progress connection event has been reported.
+        /// </summary>
+        private void OnReportProgress(Progress progress)
```

Comment:
Nit: making it `protected virtual` will let derived classes handle progress report without the need of subscribing to the event.